### PR TITLE
Edit remote repo

### PR DESCRIPTION
**Pull request by ShaiahWren**
Issue: https://issues.redhat.com/browse/AAH-618

Moving 'edit remote repo' testing from https://github.com/ansible/ansible-hub-ui/pull/1833 to a separate pr. 

Noting that there is very limited error handling with regard to input length, spaces, etc on all inputs except the URL.

### CODE REVIEW

**Comment by himdel** [comment]
is this about https://github.com/ansible/ansible-hub-ui/pull/1833#discussion_r871822962 ?

If so, that comment was just about testing the "registry" part of the form in the registry tests.

If not .. what's the difference here?

**Comment by ShaiahWren** [comment]
@himdel I moved the 'edit' portion of the repo management pr over here, per @ZitaNemeckova's request.

**Comment by himdel** [comment]
Ah, so just splitting a big pr, makes sense :)

LGTM, except this seems to break the same task* tests

**Comment by himdel** [comment]
..although, is there one type missing?

the form works in 4 modes .. `certified`, `community`, none and `registry` .. registry is handled in #1833, but I can only see `community` and `certified` here?

File: test/cypress/integration/remote_repo_edit.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+    // verify values have been saved properly.
+    cy.get('[aria-label="Actions"]:first').click(); // click the kebab menu on the 'community' repo
+    cy.contains('Edit').click();
+
+    cy.contains('Show advanced options').click();
+    cy.get('input[id="username"]').should('have.value', 'test');
+    cy.get('input[id="proxy_url"]').should('have.value', 'https://example.org');
+    cy.get('input[id="proxy_username"]').should('have.value', 'test');
+    cy.intercept(
```
Comment (ZitaNemeckova) [inline test/cypress/integration/remote_repo_edit.js:52]: No need to define it again. You could use `@editCommunityRemote` again.

File: test/cypress/integration/remote_repo_edit.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    cy.get('[aria-label="Actions"]:first').click(); // click the kebab menu on the 'community' repo
+    cy.contains('Edit').click();
+    cy.contains('Show advanced options').click();
+    cy.get('input[id="username"]').clear();
+    cy.get('input[type="password"]')
+      .siblings('button')
+      .click({ multiple: true });
+    cy.get('input[id="proxy_url"]').clear();
+    cy.get('input[id="proxy_username"]').clear();
+    cy.contains('Save').click();
```
Comment (ZitaNemeckova) [inline test/cypress/integration/remote_repo_edit.js:69]: There's no check the save was a success?

File: test/cypress/integration/remote_repo_edit.js [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    cy.get('[aria-label="Actions"]').should('be.visible');
+    cy.get('[aria-label="Actions"]').eq(1).click(); // click the kebab menu on the 'rh-certified' repo
+    cy.contains('Edit').click();
+    cy.contains('Show advanced options').click();
+    cy.get('input[id="username"]').should('have.value', 'test');
+    cy.get('input[id="proxy_url"]').should('have.value', 'https://example.org');
+    cy.get('input[id="proxy_username"]').should('have.value', 'test');
+    cy.intercept(
+      'PUT',
+      Cypress.env('prefix') + 'content/community/v3/sync/config/',
```
Comment (ZitaNemeckova) [inline test/cypress/integration/remote_repo_edit.js:112]: Is it intentional that it's a `community` URL?

**Comment by ZitaNemeckova** [comment]
@ShaiahWren Great job! I added few comments.

File: test/cypress/integration/remote_repo_edit.js (hunk 3 above)
Comment (ShaiahWren) [reply test/cypress/integration/remote_repo_edit.js:112]: Yes - any reason it can't be?

File: test/cypress/integration/remote_repo_edit.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    cy.get('[aria-label="Actions"]').eq(1).click(); // click the kebab menu on the 'rh-certified' repo
+    cy.contains('Edit').click();
+    cy.contains('Show advanced options').click();
+    cy.get('input[id="username"]').clear();
+    cy.get('input[type="password"]')
+      .siblings('button')
+      .click({ multiple: true });
+    cy.get('input[id="proxy_url"]').clear();
+    cy.get('input[id="proxy_username"]').clear();
+    cy.contains('Save').click();
```
Comment (ZitaNemeckova) [inline test/cypress/integration/remote_repo_edit.js:125]: There's no check that the save is a success. There should be a bit of code that checks that `username`, `proxy_url`, `proxy_username` are empty after the save.

**Review by ZitaNemeckova — APPROVED** [review]
LGTM 👍

**Comment by ShaiahWren** [comment]
Pr closed.

**Pull request merged**